packages/ui/src/ui/hocs/withBlockedNavigation.js

```diff
@@ -33,3 +30,16 @@ export default function withBlockedNavigation(Component) {
 
     return connect(null, {addNavigationBlocker, removeNavigationBlocker})(ResComponent);
 }
+
+export function BlockNavigation() {
```

Comment:
At its core, it is not a component. Because it doesn't have any visual component. I suggest moving this to an action or hook

Reply:
It is impossible to use hooks from class-components, so we don't have any choice here.